File: Web/src/models/responses/FromToResponse.ts
import { BaseResponse, EResponse } from "./BaseResponse";

export class FromToResponse extends BaseResponse {
  public fromTo: string[] = [];

  public defaultFrom: string = "";
  public defaultFromValue: string = "";

  public defaultTo: string = "";
  public defaultToValue: string = "";

  public constructor(fromTo?: string[]) {
    super(EResponse.FromTo);

    if (fromTo) {
      this.fromTo = fromTo;
    }
  }

  public override Initialize(data: any) {
    this.fromTo = Array.from(data.fromTo);

    this.defaultFrom = data.defaultFrom;
    this.defaultFromValue = data.defaultFromValue;

    this.defaultTo = data.defaultTo;
    this.defaultToValue = data.defaultToValue;
  }
}

File: Shared/Responses/FromToResponse.cs
﻿namespace Shared.Responses
{
public class FromToResponse
() : BaseResponse(EType.FromTo)
{
    public required List<string> FromTo { get; init; }

    public required string DefaultFrom { get; init; }
    public required string DefaultFromValue { get; init; }

    public required string DefaultTo { get; init; }
    public required string DefaultToValue { get; init; }
}
}
